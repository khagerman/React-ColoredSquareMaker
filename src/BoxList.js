import React, { useState } from "react";
import NewBoxForm from "./NewBoxForm";
import Box from "./Box";
import { v4 as uuid } from "uuid";
import "./BoxList.css";
const BoxList = () => {
  const [boxes, setBoxes] = useState([]);

  const renderBoxes = () => {
    return (
      <div className="BoxList-con">
        {boxes.map((box) => (
          <Box
            key={box.id}
            id={box.id}
            color={box.color}
            height={box.height}
            width={box.width}
            deleteBox={deleteBox}
          />
        ))}
      </div>
    );
  };

  const addBox = (box) => {
    let newBox = { ...box, id: uuid() };
    setBoxes((boxes) => [...boxes, newBox]);
  };
  const deleteBox = (id) => {
    setBoxes(boxes.filter((b) => b.id !== id));
    console.log(boxes);
    console.log(id);
  };
  return (
    <div className="BoxList">
      <NewBoxForm addBox={addBox} />
      {renderBoxes()}
    </div>
  );
};

export default BoxList;
