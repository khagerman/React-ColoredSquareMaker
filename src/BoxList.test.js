import React from "react";
import { render, fireEvent, queryByAttribute } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", function () {
  render(<BoxList />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it("can add a new square", function () {
  const { getByLabelText, queryByText } = render(<BoxList />);

  // no square yet
  expect(queryByText("x")).not.toBeInTheDocument();

  const widthInput = getByLabelText("Width:");
  const heightInput = getByLabelText("Height:");
  const colorInput = getByLabelText("Color:");

  const submitBtn = queryByText("Make a square");

  // fill out the form
  fireEvent.change(widthInput, { target: { value: 200 } });
  fireEvent.change(heightInput, { target: { value: 200 } });
  fireEvent.change(colorInput, { target: { value: "#9932CC" } });
  fireEvent.click(submitBtn);

  // item exists!
  expect(queryByText("x")).toBeInTheDocument();
});
