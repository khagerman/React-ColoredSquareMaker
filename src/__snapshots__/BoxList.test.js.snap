// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <form>
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        name="height"
        type="number"
        value="100"
      />
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        type="number"
        value="100"
      />
      <label
        for="color"
      >
        Color:
      </label>
      <input
        id="color"
        name="color"
        type="color"
        value="#7FFFD4"
      />
      <button>
        Make a square
      </button>
    </form>
    <div
      class="BoxList-con"
    />
  </div>
</DocumentFragment>
`;
